src/ext/porcelain.lisp

```diff
@@ -0,0 +1,345 @@
+
+(defpackage :porcelain
+  (:use :cl)
+  (:shadow :push)
+  (:export))
```

Comment:
This package is nice because it appears to have a good separation of dependencies and separate responsibilities :+1: 

Within the lem repository, the package name prefix is prefixed with "lem" to avoid name conflicts with all other repositories.
What about lem/porcelain etc?
It might be a good idea to use local-nickname where this package is used.

Reply:
It would be great to have an explanation of what this package is in the defpackage documentation.